File: app/src/main/java/com/material/androidpro/utils/helpers/ExceptionHelper.java
package com.material.androidpro.utils.helpers;

public class ExceptionHelper {

    private static boolean EXCEPTION_ENABLED = true;

    public static void setException(boolean isLogEnabled) {
        EXCEPTION_ENABLED = isLogEnabled;
    }

    public static void printStackTrace(Exception e) {
        if (EXCEPTION_ENABLED) {
            LogHelper.e("exception", e.getMessage());
        }
    }
}

File: app/src/main/java/com/material/androidpro/utils/helpers/ExceptionHelper.kt
package com.material.androidpro.utils.helpers

object ExceptionHelper {

    private var EXCEPTION_ENABLED = true

    fun setException(isLogEnabled: Boolean) {
        EXCEPTION_ENABLED = isLogEnabled
    }

    fun printStackTrace(e: Exception) {
        if (EXCEPTION_ENABLED) {
            e.printStackTrace()
            LogHelper.e("exception", e.message)
        }
    }

    fun printStackTrace(e: Throwable) {
        if (EXCEPTION_ENABLED) {
            e.printStackTrace()
            LogHelper.e("exception", e.message)
        }
    }
}
